lib/internal/Magento/Framework/GraphQl/Query/Fields.php

```
@@ -0,0 +1,64 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Framework\GraphQl\Query;
+
+use GraphQL\Language\AST\Node;
+use GraphQL\Language\AST\NodeKind;
+
+/**
+ * This class holds a list of all queried fields and is used to enable performance optimization for schema loading.
+ */
+class Fields
+{
+    /**
+     * @var string[]
+     */
+    private $fieldsUsedInQuery = [];
+
+    /**
+     * Prints AST to string. Capable of printing GraphQL queries and Type definition language.
+     *
+     * @param string $query
+     * @return void
+     */
+    public function setQuery($query)
+    {
+        $queryFields = [];
+        try {
+            $queryAst = \GraphQL\Language\Parser::parse(new \GraphQL\Language\Source($query ?: '', 'GraphQL'));
+            \GraphQL\Language\Visitor::visit(
+                $queryAst,
+                [
+                    'leave' => [
+                        NodeKind::NAME => function (Node $node) use (&$queryFields) {
+                            $queryFields[$node->value] = $node->value;
+                        }
+                    ]
+                ]
+            );
+        } catch (\Exception $e) {
```

Why do we need empty catch?